Patch for MDEV-39691 from this upstream PR:
https://github.com/MariaDB/server/pull/5104/changes/0a66c466f1772945d46b12bd340e1bd9adff4e9b



From 0a66c466f1772945d46b12bd340e1bd9adff4e9b Mon Sep 17 00:00:00 2001
From: Daniel Black <daniel@mariadb.org>
Date: Thu, 21 May 2026 15:08:01 +1000
Subject: [PATCH] MDEV-39691 wsrep.cnf contains deprecated wsrep_causal_reads

wsrep_causal_reads was deprecated in 10.1.3 and removed in 11.3.1.
Put in place the equivalent (and default) wsrep_sync_wait=0.

Amend a few early comments to show MariaDB as our product.
---
 support-files/wsrep.cnf.sh | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/support-files/wsrep.cnf.sh b/support-files/wsrep.cnf.sh
index 7242fff9f7ac0..54d67d951d3b2 100644
--- a/support-files/wsrep.cnf.sh
+++ b/support-files/wsrep.cnf.sh
@@ -87,8 +87,9 @@ wsrep_auto_increment_control=1
 # retry autoinc insert, which failed for duplicate key error
 wsrep_drupal_282555_workaround=0
 
-# enable "strictly synchronous" semantics for read operations
-wsrep_causal_reads=0
+# Syncronization causality setting, (0 - disabled)
+# same as deprecated and removed wsrep_causal_reads=0.
+wsrep_sync_wait=0
 
 # Command to call when node status or cluster membership changes.
 # Will be passed all or some of the following options:
